routes/registration.js

```diff
@@ -14,42 +14,76 @@ const validateEmail = utilityFunctions.validateEmail;
 const validateName = utilityFunctions.validateName;
 const validatePassword = utilityFunctions.validatePassword;
 const randomStr = utilityFunctions.randomString;
+const readHTMLFile = utilityFunctions.readHTMLFile;
+const fs = require('fs');
+
+const codes = JSON.parse(fs.readFileSync('./data/codes.json'));
+const errorCode = codes.errors;
+const successCode = codes.success;
+
+const otherData = JSON.parse(fs.readFileSync('./data/english.json'));
+const mailData = otherData.mail;
+const mailFooterData = otherData.mail.footer;
+const mailHeaderData = otherData.mail.header;
 
 function verificationMail(req, res, rString) {
-    mail.mailOptions.to = req.body.email;
-    mail.mailOptions.subject = 'PC PrepKit Email Verification';
-    mail.mailOptions.html = `<b>Click on the link to complete the verification</b> <a href='${config.basePath}verification?token=${rString}&user=${req.body.email}'>Verify</a>`;
-    mail.smtpTransport.sendMail(mail.mailOptions, function(error) {
-        if (error) {
-            res.status(500).json({error: 'Something Went Wrong! Try again later.'});
-        } else {
-            res.json('Verification Mail Sent, Please check your mail.');
-        }
+    readHTMLFile('./public/pages/registration-verification.html', function(err, html) {
+        const template = handlebars.compile(html);
+        const replacements = {
```

Comment:
I tried generalising it, but there are too many parameters to pass to the generalised function and they are different for different cases, so I decided not to go ahead with the generalising.